mlflow/utils/requirements_utils.py

```diff
@@ -442,6 +442,12 @@ def _check_requirement_satisfied(requirement_str):
     Otherwise, returns an instance of `_MismatchedPackageInfo`.
     """
     _init_packages_to_modules_map()
+
+    if re.match(r"pyspark[^\w-]", requirement_str) and is_in_databricks_runtime():
+        # for databricks runtime pyspark, ignore micro-version mismatch.
+        # replace == operator with ~= operator, but keep === operator unchanged.
+        requirement_str = re.sub("([^=])==([^=])", r"\1~=\2", requirement_str)
```

Comment:
Can this handle a requirement string that looks like `pyspark>=1.2.3,==1.2.*`?

Reply:
emm. this is an issue.

Reply:
i will update regex to cover the case ==3.2 or ==3.2.*

Reply:
I don't think it needs to, right? The purpose of this change is to make sure that exact requirement specifiers generated automatically by MLflow, e.g. `pyspark==3.2.1`, are not interpreted too strictly on Databricks. If the user wants to supply a custom / stricter specifier, then they should be able to without us interfering / interpreting the specifier differently.

Reply:
I'm trying to come up with a more robust approach:

```python
import pkg_resources
from packaging.requirements import Requirement
from packaging.specifiers import SpecifierSet
import re


def is_satisfied(req, version):
    req = pkg_resources.Requirement.parse(req)
    if req.name == "pyspark" and len(req.specifier) == 1:
        m = re.match(r"==([^=]+)", str(req.specifier))
        if m:
            return SpecifierSet(f"~={m.group(1)}").contains(version)
    return req.specifier.contains(version)


print(is_satisfied("pyspark == 1.2.2", "1.2.3"))  # True
print(is_satisfied("pyspark>=1.2.3,==1.2.*", "1.2.3"))  # True
print(is_satisfied("pyspark === abc", "1.2.3"))  # False
print(is_satisfied("scikit-learn == 1.2.2", "1.2.3"))  # False

```

Reply:
Actually, @WeichenXu123 , do we need this PR? The micro version isn't going to change within the same MLR / DBR version, right?

Reply:
Confirmed with @hyukjin  the pyspark version is not changed within the same MLR/DBR version